### ISSUE TITLE
Sources.fileWatcher re-emits all the lines on Linux

### ISSUE THREAD

**Issue by vladoschreiner** [hazelcast/hazelcast-jet#1323]
The fileWatcher is supposed to emit new lines added to a file in a specified directory.

On Linux, it emits all the lines in a file every time the line is added and the file is saved. It works properly on macOS.

**Comment by viliam-durina** [comment]
Which editor did you use? I tried `vi` editor and it has weird behavior: it adds a `\n` at the end of the file without displaying it. So if you add `line1\n`, the file will actually contain `line1\n\n`. If you save the file, you'll see two items emitted by the file watcher: `line1` and an empty string. If you continue typing, for example you add `line2\n`, vi will delete the last newline and turn the file contents to `line1\nline2\n\n`. And because the newline was deleted and replaced with another character, the source will emit two more items: `ine2` and another empty string. This is expected because the source remembers the offset up to which it read the file for the last time and that offset was after the second `\n`, which got replaced by `l`, which is missing, therefore you see `ine2` and not `line2`.

I suspect your editor actually deletes the file and then writes it in full. The source detects the deletion and removes the offset. When it's written in full again, all lines get emitted again. Try appending lines with `echo line >> yourfile`, that should work.
